packages/iris-grid/src/AdvancedFilterCreator.tsx

```diff
@@ -121,19 +121,17 @@ class AdvancedFilterCreator extends PureComponent<
     this.handleUpdateTimeout = this.handleUpdateTimeout.bind(this);
 
     this.focusTrapContainer = React.createRef();
-    this.filterKey = 0;
 
     const { options } = props;
     let { filterOperators, invertSelection, selectedValues } = options;
 
-    let filterItems: AdvancedFilterItem[] = options.filterItems.map(
-      ({ selectedType, value }) => ({
+    let filterItems: AdvancedFilterItem[] =
+      options.filterItems?.map(({ selectedType, value }) => ({
         selectedType,
         value,
         key: shortid(),
-      })
-    );
-    if (filterItems == null) {
+      })) ?? [];
```

Comment:
`let filterItems: AdvancedFilterItem[] =
      options.filterItems?.map(({ selectedType, value }) => ({
        selectedType,
        value,
        key: shortid(),
      })) ?? [AdvancedFilterCreator.makeFilterItem()];`

Then you don't need thee `filterItems.length === 0` check

Reply:
this

Reply:
Tried, but actually while it can be initialized as null, but if you set something then remove it, then next time it is initialized as [], so you still need the === 0 check.